test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/MacroProcessorTests.cs

```diff
@@ -27,92 +28,204 @@ public void CanThrow_WhenCannotProcessMacro()
+                virtualize: true, environment: A.Fake<IEnvironment>(), addLoggerProviders: new[] { loggerProvider });
 
-            MacroProcessingException e = Assert.Throws<MacroProcessingException>(() => MacroProcessor.ProcessMacros(engineEnvironmentSettings, config, new VariableCollection()));
-            Assert.Equal("Failed to evaluate", e.InnerException?.Message);
+            var macroConfigs = new[] { new FakeMacroConfig(new FakeMacro(), "testVariable", "dummy") };
+            MacroProcessor.ProcessMacros(engineEnvironmentSettings, macroConfigs, new VariableCollection());
+
+            Assert.True(loggedMessages.Count == 1);
+            Assert.Equal("Generated symbol 'testVariable': type 'fake' is unknown, processing is skipped.", loggedMessages.First().Message);
         }
 
         [Fact]
-        public void CanThrow_WhenAuthoringError()
+        public void CanProcessMacroWithCustomMacroAsDependency()
         {
-            IEngineEnvironmentSettings engineEnvironmentSettings = _environmentSettingsHelper.CreateEnvironment(virtualize: true, additionalComponents: new[] { (typeof(IGeneratedSymbolMacro), (IIdentifiedComponent)new FailConfigMacro()) });
-
-            GlobalRunConfig config = new()
+            var fakeMacroVariableName = "testVariable";
+            List<(LogLevel Level, string Message)> loggedMessages = new();
+            InMemoryLoggerProvider loggerProvider = new(loggedMessages);
+            var generatedConfig = A.Fake<IGeneratedSymbolConfig>();
+            A.CallTo(() => generatedConfig.Parameters).Returns(new Dictionary<string, string>()
             {
-                GeneratedSymbolMacros = new[] { new GeneratedSymbol("s", "fail") }
-            };
+                { "sourceVariableName",  JExtensions.ToJsonString("dummy") },
+                { "fallbackVariableName",  JExtensions.ToJsonString(fakeMacroVariableName) }
+            });
+            A.CallTo(() => generatedConfig.VariableName).Returns("coalesceTest");
+
+            var coalesceMacroConfig = new CoalesceMacroConfig(new CoalesceMacro(), generatedConfig);
+            coalesceMacroConfig.Dependencies.Add(fakeMacroVariableName);
+
+            var engineEnvironmentSettings = _environmentSettingsHelper.CreateEnvironment(virtualize: true, environment: A.Fake<IEnvironment>(), addLoggerProviders: new[] { loggerProvider });
+
+            var fakeMacro = new FakeMacro();
+            var variableCollection = new Dictionary<string, object>() { { fakeMacroVariableName, fakeMacro } };
+
+            MacroProcessor.ProcessMacros(engineEnvironmentSettings, new[] { coalesceMacroConfig }, new VariableCollection(default, variableCollection));
 
-            TemplateAuthoringException e = Assert.Throws<TemplateAuthoringException>(() => MacroProcessor.ProcessMacros(engineEnvironmentSettings, config, new VariableCollection()));
-            Assert.Equal("bad config", e.Message);
+            Assert.True(variableCollection.Count == 2);
```

Comment:
Please assert the variable collection for correct macro values

Reply:
done